### ISSUE TITLE
`curl: (56) Proxy CONNECT aborted` error with px 0.8 when using curl.exe

### ISSUE THREAD

**Issue by s-t-hg** [genotrance/px#174]
px 0.7 works fine, but when I start 0.8 and using curl.exe (with the -x, --proxy option) I get this error and it doesn't work properly. (using binaries for Windows).
I don't know what's going on because nothing is logged even if I start it with the debug option.

**Comment by genotrance** [comment]
Can you provide some info on how Px is configured? Perhaps the startup banner and settings?

**Comment by s-t-hg** [comment]
Thanks for the reply.
I put the px folder to `C:\Users\username\AppData\Local\Programs\px`, and registered px to the start up with the `./px.exe --install` command.
The `px.ini` file is like the following.

```ini
[proxy]
server = 
pac = http://example.com
port = 3128
listen = 127.0.0.1
allow = *.*.*.*
gateway = 0
hostonly = 1
noproxy = 192.168.65.0/28
useragent = 
username = 
auth = 

[settings]
workers = 2
threads = 5
idle = 30
socktimeout = 20.0
proxyreload = 60
foreground = 0
log = 0

```

**Comment by genotrance** [comment]
Have you already tried setting `log = 1` in the ini file?

Also, can you run `reg query HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Run /v Px` and share the output?

Lastly, please also share the `curl` command like example you are running. Thanks!

**Comment by s-t-hg** [comment]
Sure, I tried px-v0.8.3 and this it the test command on pwsh and the px's log. (IP addresses, etc. have been replaced.)

```sh
$ curl -I https://example.com/ --proxy http://localhost:3129
curl: (56) Proxy CONNECT aborted
```

```log
MainProcess: MainThread: 1670895787: /main/parse_config/__init__: Proxy mode = MODE_CONFIG_PAC
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: libcurl/7.83.1 OpenSSL/3.0.2 (Schannel) zlib/1.2.12 brotli/1.0.9 libidn2/2.3.2 libssh2/1.10.0 nghttp2/1.47.0 ngtcp2/0.5.0 nghttp3/0.4.1 libgsasl/1.10.0
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_SSL: True
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_SSPI: True
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_SPNEGO: True
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_GSSAPI: False
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_GSSNEGOTIATE: False
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_KERBEROS5: True
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_NTLM: True
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: CURL_VERSION_NTLM_WB: False
MainProcess: MainThread: 1670895787: /parse_config/__init__/print_curl_version: Host: x86_64-pc-win32
Serving at :3128 proc MainProcess
MainProcess: MainThread: 1670895787: /run_pool/print_banner/detach_console: Freed console successfully
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:server = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:pac = http://example.com:8080
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:port = 3128
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:listen = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:allow = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:gateway = 0
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:hostonly = 1
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:noproxy = 192.168.65.0/28
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:useragent = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:username = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: proxy:auth = 
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:workers = 2
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:threads = 5
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:idle = 30
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:socktimeout = 20.0
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:proxyreload = 60
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:foreground = 0
MainProcess: MainThread: 1670895787: /main/run_pool/print_banner: settings:log = 1
MainProcess: MainThread: 1670895804: /serve_forever/_handle_request_noblock/verify_request: Client address: 127.0.0.1
MainProcess: MainThread: 1670895804: /serve_forever/_handle_request_noblock/verify_request: Host-only IP allowed
MainProcess: Thread_0: 1670895804: /do_CONNECT/do_curl/__init__: da3f93817ccbd41773b19002926cad1d1cde4b31: New curl instance
MainProcess: Thread_0: 1670895804: /do_curl/__init__/_setup: da3f93817ccbd41773b19002926cad1d1cde4b31: CONNECT example.com:443 using HTTP/1.1
MainProcess: Thread_0: 1670895804: /handle_one_request/do_CONNECT/do_curl: da3f93817ccbd41773b19002926cad1d1cde4b31: Path = example.com:443
MainProcess: Thread_0: 1670895804: /find_proxy_for_url/find_proxy_for_url/get_netloc: netloc = ('example.com', 443), path = /
MainProcess: Thread_0: 1670895804: /find_proxy_for_url/find_proxy_for_url/__init__: Loading PAC utils
MainProcess: Thread_0: 1670895804: /find_proxy_for_url/find_proxy_for_url/load_url: Loading PAC url: http://example.com:8080
MainProcess: Thread_0: 1670895804: /find_proxy_for_url/load_url/__init__: d174f4f25758d669e4393c6c76fcd343bc3e445d: New curl instance
MainProcess: Thread_0: 1670895804: /load_url/__init__/_setup: d174f4f25758d669e4393c6c76fcd343bc3e445d: GET http://example.com:8080 using HTTP/1.1
MainProcess: Thread_0: 1670895804: /find_proxy_for_url/load_url/buffer: d174f4f25758d669e4393c6c76fcd343bc3e445d: Setting up buffers for bridge
MainProcess: Thread_0: 1670895804: /load_url/buffer/bridge: d174f4f25758d669e4393c6c76fcd343bc3e445d: Setting up bridge
MainProcess: Thread_0: 1670895804: /perform/do/add: d174f4f25758d669e4393c6c76fcd343bc3e445d: Handles = 0
MainProcess: Thread_0: 1670895804: /do/add/_add_handle: d174f4f25758d669e4393c6c76fcd343bc3e445d: Add handle
MainProcess: Thread_0: 1670895804: /do/add/_add_handle: d174f4f25758d669e4393c6c76fcd343bc3e445d: Added handle
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Curl info: Trying XXX.XXX.XXX.XXX:8080...
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Curl info: Connected to example.com (XXX.XXX.XXX.XXX) port 8080 (#0)
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Sent header => GET / HTTP/1.1
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Sent header => Host: example.com:8080
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Sent header => Accept: */*
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Curl info: Mark bundle as not supporting multiuse
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= HTTP/1.1 200 OK
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Date: Tue, 13 Dec 2022 01:43:24 GMT
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Server: Apache/2.2.17 (Unix) mod_ssl/2.2.17 OpenSSL/0.9.8e-fips-rhel5
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Last-Modified: Tue, 22 Nov 2022 08:40:55 GMT
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= ETag: "50054c-8f34-5ee0b2157b3c0"
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Accept-Ranges: bytes
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Content-Length: 36660
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Received header <= Content-Type: application/x-ns-proxy-autoconfig
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_header_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Done sending headers
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_write_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Wrote 6998 bytes
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_write_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Wrote 16384 bytes
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_write_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Wrote 13278 bytes
MainProcess: Thread_0: 1670895804: /_perform/_socket_action/_debug_callback: d174f4f25758d669e4393c6c76fcd343bc3e445d: Curl info: Connection #0 to host example.com left intact
MainProcess: Thread_0: 1670895804: /perform/remove/_remove_handle: d174f4f25758d669e4393c6c76fcd343bc3e445d: Remove handle: 
----------------------------------------
Exception occurred during processing of request from ('127.0.0.1', 63632)
Traceback (most recent call last):
  File "C:\Users\username\AppData\Local\Programs\px\socketserver.py", line 683, in process_request_thread
  File "C:\Users\username\AppData\Local\Programs\px\socketserver.py", line 360, in finish_request
  File "C:\Users\username\AppData\Local\Programs\px\socketserver.py", line 747, in __init__
  File "C:\Users\username\AppData\Local\Programs\px\http\server.py", line 425, in handle
  File "C:\Users\username\AppData\Local\Programs\px\px\main.py", line 295, in handle_one_request
  File "C:\Users\username\AppData\Local\Programs\px\http\server.py", line 413, in handle_one_request
  File "C:\Users\username\AppData\Local\Programs\px\px\main.py", line 392, in do_CONNECT
  File "C:\Users\username\AppData\Local\Programs\px\px\main.py", line 325, in do_curl
  File "C:\Users\username\AppData\Local\Programs\px\px\main.py", line 399, in get_destination
  File "C:\Users\username\AppData\Local\Programs\px\px\wproxy.py", line 465, in find_proxy_for_url
  File "C:\Users\username\AppData\Local\Programs\px\px\wproxy.py", line 256, in find_proxy_for_url
  File "C:\Users\username\AppData\Local\Programs\px\px\pac.py", line 60, in find_proxy_for_url
_quickjs.JSException: InternalError: Python call failed.
    at FindProxyForURL (<input>:24)

----------------------------------------
```

Maybe the same problem of #161?

**Comment by genotrance** [comment]
Very likely! Thanks for taking the time to test this out. I'll get a fix out as soon as possible.

**Comment by genotrance** [comment]
This should be fixed in v0.8.4.
